### DISCUSSION TITLE
Lesson 7  ( error: a value is required for '--fork-url <URL>' but none was supplied)

### DISCUSSION THREAD

**Discussion by 0xtdey** [Cyfrin/foundry-full-course-cu#216] [Q&A] [answered]
<img width="650" alt="image" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/91402167/ef17a303-38e3-4a9e-b973-496545ef12cc">

Why am I getting this error?

**Comment by devrajsinghrawat** [comment]
1 - Make sure SEPOLIA_RPC_URL value is maintained in .env file

2 - run the command source .env in the command prompt 

3 - verify that the value is exported by echo $SEPOLIA_RPC_URL

and then run this command

**Reply by DukeLurkin** [reply]
Had to run source .env in command, thx

**Reply by Gamal-Omar** [reply]
Where is the .env file located?

**Comment by Aishwaria8786** [comment, marked as answer]
source .env will solve the problem.

**Reply by 0xtdey** [reply]
source .env wasn't working earlier, reopened the project and ran again it worked. Thanks !

**Reply by sebghatullahmalikzay** [reply]
i had a similar issue on Ubuntu OS but running source.env resolved the issue. thanks a lot

**Reply by ShawnFromTexas** [reply]
fixed my issue also. many thanks everyone!

**Reply by viveknath13** [reply]
hey,
 how did you slove it.env issue

**Reply by nicoarbio** [reply]
worked on windows 10. thx!

**Reply by EngrPips** [reply]
Interesting.

**Reply by Gamal-Omar** [reply]
Where is the .env file located?

**Reply by EngrPips** [reply]
> Where is the .env file located?

It should be at the root directory of your codebase.

**Reply by Gamal-Omar** [reply]
> > Where is the .env file located?
> 
> It should be at the root directory of your codebase.

it was not downloaded,  but after a lot of research i download this file "dotenv-cli": "^7.4.2", but the problem is still there

**Reply by nicoarbio** [reply]
> > > Where is the .env file located?
> > 
> > 
> > It should be at the root directory of your codebase.
> 
> it was not downloaded, but after a lot of research i download this file "dotenv-cli": "^7.4.2", but the problem is still there

The `.env` file has environment variables that foundry needs to work. You should configure this file with eg: `SEPOLIA_RPC_URL="<your_url>"` and then give this context to your current terminal using `source .env` (wherever .env is located, adding the full filepath if needed)

**Reply by EngrPips** [reply]
Good heads-up.

**Reply by Gamal-Omar** [reply]
Whoever faces that problem should check the course at 11:13:34

**Reply by Gamal-Omar** [reply]
> > > > Where is the .env file located?
> > > 
> > > 
> > > It should be at the root directory of your codebase.
> > 
> > 
> > it was not downloaded, but after a lot of research i download this file "dotenv-cli": "^7.4.2", but the problem is still there
> 
> The `.env` file has environment variables that foundry needs to work. You should configure this file with eg: `SEPOLIA_RPC_URL="<your_url>"` and then give this context to your current terminal using `source .env` (wherever .env is located, adding the full filepath if needed)

Thanks, it helped a lot.

**Reply by EngrPips** [reply]
Awesome.

**Comment by Mindsetter** [comment]
I had a similar issue on Windows OS but running ```source .env``` resolved the issue. Thank you

**Comment by hagarabobakr** [comment]
1-run the command source .env in the command prompt
2-verify that the value is exported by echo $RPC_URL and $PRIVATE_KEY
 If you did these steps and the problem still exists, make sure that you saved the .env file

**Comment by sepideh54** [comment]
I also got this error, but because Alchemy has restricted my country, is there another site where I can get api without such a restriction?

**Reply by vigneshbalu96** [reply]
which country does Alchemy got restricted? @sepideh54

**Comment by vigneshbalu96** [comment]
Work Underhood : 
Reason For Error : Before running `source .env` the shell session doesn't reads the .env file and not sets the environment variables defined in it. 
Solution : By running `source .env` the shell session reads the .env file and sets the environment variables defined in it. 
Benefit : These variables can then be accessed by any command or script run in the same shell session. This practice helps keep sensitive information out of your codebase and allows us to easily change configuration values without modifying the scripts.
